### DISCUSSION TITLE
Imagus sieves support

### DISCUSSION THREAD

**Discussion by extesy** [extesy/hoverzoom#992] [Ideas]
Hey @GrosPoulet, have you by any chance looked at how imagus processes its sieves? It seems like a combination of one or two scripts and some regexps. What do you think about possibility of adding a plugin to HZ+ that would natively support those sieves?

**Comment by GrosPoulet** [comment]
Sounds like a good idea, but not so easy to implement.
Tasks:
- in Options: enable/disable sieve(s), as we do with HZ+ plug-in(s)
- write a generic Imagus plug-in that will apply all sieves to pages (i don't know exactly how Imagus handles its sieves, but all sieves are applied to any url)
- handle results return by Imagus in HZ+ main script (simply add them to data().hoverZoomSrc? add a distinct storage?)
- ...

BTW, i always look at sieves for inspiration when writing a plug-in (like the one for imagefap).
Maybe you should just find a way to turn a sieve into a HZ+ plug-in automatically ?

**Reply by extesy** [reply]
> i don't know exactly how Imagus handles its sieves, but all sieves are applied to any url

That would work similar to all `*_a.js` plugins that we have today - they are also applied to all urls. There might be some conflicts, though, between sieves and plugins.

> Maybe you should just find a way to turn a sieve into a HZ+ plug-in automatically?

Yes, this should be possible, but the main benefit of imagus sieves is that they can be added or updated directly by end users without a need to wait for me to publish a new extension version to webstores. There are even [communities](https://forum.ru-board.com/topic.cgi?forum=5&topic=50874&start=0&limit=1&m=5#1) that maintain these sieves similar to how ad block lists are maintained separately from ad block extensions.

I'll try to find some time to look at imagus source code and will share my findings. Hope you can do that too, but no pressure - this is a very aspirational idea, I don't even know if these two approaches are compatible or must be mutually exclusive.

**Reply by LiliaDoe** [reply]
I had a thought on this. While being able to directly import sieves would be ideal, allowing a similar functionality may be much easier. By that, I mean adding a way for users to directly add new plugins on their end. It would probably help to update the storage system so the settings wouldn't be removed whenever users clear all their site data. I haven't personally tested this, but users have reported this issue.

It won't allow direct one to one importing of sieves, but it'll give the main benefit of sieves.

**Reply by extesy** [reply]
I think using the existing syntax for user-generated plugins would be too hard (for users). People who are able to write plugins directly in javascript, are also able to send pull requests for this repo. Two main benefits of imagus sieves are: 1) simple syntax, 2) existing community.

**Reply by LiliaDoe** [reply]
Gotcha. I still think allowing users to update plugins on their end without having to fork the repo would be useful, but not a priority.
